### ISSUE TITLE
CommentNode#getTotalSize() is possibly flawed

### ISSUE THREAD

**Issue by Saketme** [mattbdean/JRAW#189]
I was trying to show the number of child comments when a comment is collapsed and noticed that the value returned by `CommentNode#getTotalSize()` is wrong. Here's the code:

```java
public int getTotalSize() {
  int size = getImmediateSize();
  for (CommentNode node : walkTree()) {
    size += node.getImmediateSize();
  }
  return size;
}
```

For a comment that has 1 child, this method correctly returns 2 as the total size. However for a comment with no child, this method returns 0 which is incorrect.

**Comment by thatJavaNerd** [comment]
That's strange. `getImmediateSize()` should get the amount of direct children the node has, so a comment with 1 child should return `1` as seen from the code:

```java
public int getImmediateSize() {
    return children.size();
}
```

Here's the javadoc for `getTotalSize()`:

>Gets the total number of children this node has. Will include this node's immediate size, the immediate size of every child, and so on.

How did you get `getTotalSize()` to return `2` for a CommentNode with 1 child?

**Comment by Saketme** [comment]
Sorry for replying this late. 

> That's strange. getImmediateSize() should get the amount of direct children the node has, so a comment with 1 child should return 1 as seen from the code:

That's correct. The source of bug here is that the `CommentNode`, on which `getTotalSize()` is called, gets referenced twice. One before the `for` loop at line #593 and second time when `walkTree()` starts at the same node before reaching its children. 

Does this make sense?

**Comment by thatJavaNerd** [comment]
I see what you're saying. This should be fixed in 1.0.0

**Comment by mattbdean** [comment]
Commit [`8ff365d`](https://github.com/mattbdean/JRAW/commit/8ff365d89a3f766b9fd828df9944bcbdbee7e53d) fixes this issue.

The new code looks like this:

```kotlin
override fun walkTree(order: TreeTraverser.Order): List<CommentNode<*>> {
    // ...
}

override fun totalSize(): Int {
    // walkTree() goes through this node and all child nodes, but we only care about child nodes
    return walkTree().size - 1
}
```
